src/collections/Users/access/byGlobalRoleOrTenantDomain.ts

```diff
@@ -8,7 +8,10 @@ import { Access, CollectionConfig } from 'payload'
 export const byGlobalRoleOrTenantDomain: (method: ruleMethod) => Access =
   (method: ruleMethod): Access =>
   (args) => {
-    if (!args.req.user) {
+    const { req } = args
```

Comment:
nit: I prefer not doing this since it can be hard to follow, and hard to ensure everyone uses the new variables - to wit, you forgot to update `args.req.payload` to `req.payload` on L24. If you prefer having more tightly scoped variables, please destructure in the `(args) => {` signature so `args` isn't in scope at all and it's not possible to make that mistake!